### ISSUE TITLE
Contact widget: Display of contacts broken for special account types

### ISSUE THREAD

**Issue by AlfredSK** [friendica/friendica#6800]
### Expected behavior
When I look at the contact widget on the profile page of a forum or news account I should see the correct number of public contacts and their avatars.

### Actual behavior
The number of contacts is mostly shown as "0" or the avatars are missing.

### Steps to reproduce the problem
When you are on 2019.03-rc and you have a forum or news account on your node go to the profile page of that account and look at the contact widget.
Or see here: https://forum.friendi.ca/profile/helpers

Just in case it matters:
The Forum node is using storage filesystem and all files have been moved from DB to a local folder.
The Libranet node is using storage filesystem and the already existing files haven't been moved to a local folder. They're still stored in the DB.

![bildschirmfoto von 2019-03-03 11-30-35](https://user-images.githubusercontent.com/3656054/53694058-739dce80-3da9-11e9-9d16-8c426e90291a.png)

![bildschirmfoto von 2019-03-03 11-27-54](https://user-images.githubusercontent.com/3656054/53694060-7b5d7300-3da9-11e9-8c2b-7885f00412c0.png)

![bildschirmfoto von 2019-03-03 11-25-44](https://user-images.githubusercontent.com/3656054/53694103-3554df00-3daa-11e9-9763-6968b8c020de.png)

### Friendica version you encountered the problem
2019.03-rc

### Friendica source (git, zip)
git

### PHP version
7.2.15

### SQL version
10.1.38 MariaDB

**Comment by AlfredSK** [comment]
I want to point out that the behavior of the contact widget has been changed by @MrPetovan some time ago. IIRC the change was about only showing public contacts with bidirectional relationship. After that change we had a similar issue with that widget on forum and news accounts because they don't have bidirectional relationships with their contacts, just followers. That was fixed back then.
Maybe this issue is showing some kind of regression.

**Comment by MrPetovan** [comment]
I can reproduce this: https://friendica.mrpetovan.com/profile/friendicafr

**Comment by MrPetovan** [comment]
Ok, we have a serious semantics issue around the `contact.rel` column values and the related `Model\Contact` constants.

The current definition of those class constants is `Contact is`. Which would mean:
- `Model\Contact::FOLLOWER`: The contact is a follower of the User this contact belongs to.
- `Model\Contact::SHARING`: The contact shares their content with the User this contact belongs to.

However, this isn't the case in practice. After reviewing my own contacts who aren't mutual friends, here's what is actually enforced in `contact` rows:
- `Model\Contact::FOLLOWER`: The User this contact belongs to shares their content with the contact.
- `Model\Contact::SHARING`: The User this contact belongs to is a follower of the contact.

In other words, the definition of these constants should be `User is`.

It would be fine if only the constant definition in the comment would be wrong and the constants would be used consistently through the code, but it doesn't appear to be the case.

For example, in `api_friendship_destroy()` (used to unfollow), we either:
- Delete the `contact` row if the `rel` was just `Model\Contact::SHARING`.
- Update `rel` to `Model\Contact::FOLLOWER` if it was a mutual relationship.

These make sense in the `Contact is` case but not in the `User is`.

Am I missing something? Will I have to check every single usage of these constants?

**Comment by nupplaphil** [comment]
Fixed at #6807
